File: solutions/beecrowd/1805/1805.r
input <- readLines(con = "stdin")

for (line in input) {
  if (line == "") next

  numbers <- as.numeric(strsplit(line, " ")[[1]])
  first <- numbers[1]
  last <- numbers[2]

  result <- (last - first + 1) * (first + last) / 2
  cat(sprintf("%.0f\n", result))
}

File: solutions/beecrowd/1805/1805.rs
use std::io;

fn main() {
    let mut input = String::new();

    while io::stdin().read_line(&mut input).is_ok() {
        if input.is_empty() {
            break;
        }

        let numbers = input
            .split_whitespace()
            .map(|s| s.parse::<u64>())
            .collect::<Result<Vec<_>, _>>();

        if let Ok(nums) = numbers {
            if nums.len() == 2 {
                let first = nums[0];
                let last = nums[1];
                println!("{}", (last - first + 1) * (first + last) / 2);
            }
        }

        input.clear();
    }
}
